paper.tex

```diff
@@ -675,6 +677,31 @@ \subsection{Smart Contract}
 	\end{tabular}
 	}
 \end{table*}
+\subsection{MAM vs. the delegated MAM}
+What follows is an account of performance comparison between MAM and delegated MAM (E2EE) for different payload sizes. Elapsed time for internet communication and PoW are not included, because the following processes starting from sending message to tangle-accelerator are the common steps. Excluding the common steps allows us to compare the performance between MAM and delegated MAM more clearly. In addition, MAM Channel creation and E2EE asymmetric key pair generation are not included, too. The two excluded steps take much more time than encrypting messages only, so counting these two steps would cause extreme values.
+
+The steps counted in MAM benchmark including:
+\begin{itemize}
+	\item Encrypt input payload
+	\item Publish final message on IOTA Tangle
+\end{itemize}
+
+The steps counted in E2EE benchmark including:
+\begin{itemize}
+	\item Generate symmetric key for encryption
+	\item Hash input payload
+	\item Encrypt input payload
+	\item Transmit final message to tangle-accelerator
+\end{itemize}
+
+\begin{figure}[!t]
+    \centering
+    \includegraphics[width=3in]{mam_vs_e2ee}
+    \caption{Compare the performance between MAM and E2EE on Raspberry Pi.}
+    \label{fig:mam_vs_e2ee}
+\end{figure}
+
+On Fig.~\ref{fig:mam_vs_e2ee}, a comparison of the two result reveals elapsing time of E2EE is obviously less than elapsing time of MAM which shows the E2EE process presented in the paper is a much more appropriate solution for low-level embedded system to send authenticated streaming data. Next, regarding the trend of the two lines, the performance of E2EE is obviously more steady than MAM in the given range. The bump of MAM performance around 750 bytes proves this point. Moreover, the performance of E2EE is almost under constant time, but MAM takes linear time on encrypting messages.
```

Comment:
Use "in the figure"
Reference: https://english.stackexchange.com/questions/27689/usage-of-in-figure-vs-on-figure